CDH2.py

```diff
@@ -0,0 +1,348 @@
+from typing import Union, Tuple, Dict
+from abc import ABC, abstractmethod
+
+
+
+def analyse_gc(records: str, min_gc: Union[int, float], max_gc: Union[int, float]) -> str:
+    """
+    Return filtered FASTQ-sequences by GC-content
+    
+    :param records: FASTQ-file
+    :type records: str
+    :param min_gc: Left boundary for GC-content filtration
+    :type min_gc: Union[int, float]
+    :tupe max_gc: Right boundary for GC-content filtration
+    :type max_gc: Union[int, float]
+    :rtype: str
+    :return: filtered FASTQ-sequences
+    """
+    return {record.id: record for record in records if (min_gc/100) <= gc_fraction(record.seq) <= (max_gc/100)}
+
+
+def filter_by_length(records: str, min_length: Union[int, float], max_length: Union[int, float]) -> str:
+    """
+    Return filtered FASTQ-sequences by GC-content
+    
+    :param records: FASTQ-file
+    :type records: str
+    :param min_gc: Left boundary for GC-content filtration
+    :type min_gc: Union[int, float]
+    :tupe max_gc: Right boundary for GC-content filtration
+    :type max_gc: Union[int, float]
+    :rtype: str
+    :return: filtered FASTQ-sequences
+    """
+
+    return {record.id: record for record in records if min_length <= len(record.seq) <= max_length}
+
+
+def filter_by_quality(records: str, quality_threshold: Union[int, float]) -> str:
+    """
+    Return filtered FASTQ-sequences by quality
+    :param records: Sequnces filtered by GC-content and length
+    :type records: str
+    :param quality_threshold: boundary for quality filtration
+    :type quality_threshold: Union[int, float]
+    :rtype: str
+    :return: filtered FASTQ-sequences   
+    """
+    
+    return {record.id: record for record in records if min(record.letter_annotations["phred_quality"]) >= quality_threshold}
+
+
+
+def write_filtered_sequences_to_fastq(filtered_sequences: Dict[str, str], unfiltered_sequences: Dict[str, str],
+                                      output_file: str, folder_path: str = 'fastq_filtrator_results') -> str:
+    '''
+    The function writes filtered FASTQ reads into new file and save it into folder "fastq_filtrator_resuls"
+    :param filtered_sequences: Dict of filtered sequences by GC-content, quality and length
+    :filtered_sequences type: Dict[str]
+    :param unfiltered_sequences: Dict of unfiltered sequences by GC-content, quality and length
+    :filtered_sequences type: Dict[str]
+    :output_file: name of output file
+    :output_file type: str
+    :folder_path: name of result folder, default = 'fastq_filtrator_resuls'
+    :folder_path type: str
+    :rtype: None
+    :return: None
+    '''
+
+    if not os.path.exists(folder_path):
+        os.makedirs(folder_path)
+
+    file_path_filtered = os.path.join(folder_path, output_file)
+    file_path_not_filtered = os.path.join(folder_path, 'unfiltered_sequences.fasta')
+    
+    with open(file_path_filtered, "w") as output_handle:
+        for key, value in filtered_sequences.items():
```

Comment:
Исправил больше для себя) так понятнее, что в каждой переменной
```suggestion
        for seq_id, seq in filtered_sequences.items():
```

Reply:
Абсолютно верное исправление!